test/test_python_gen.py

```diff
@@ -1198,6 +1217,55 @@ def test_struct_decoding_with_optional_struct(self):
         self.assertEqual(s2.f1.f1, 'hello')
         self.assertEqual(s2.f1.f2, 3)
 
+    def test_struct_equality_with_object(self):
+        """Should not throw an error when comparing with object.
+
+        Object is a superclass of Struct, but it should not be considered for equality.
+        """
+        s = self.decode(self.sv.Struct(self.ns.S), json.dumps({'f': 'F'}))
+        self.assertFalse(s == object())
+
+    def test_struct_equality_with_value(self):
+        s = self.decode(self.sv.Struct(self.ns.S), json.dumps({'f': 'F'}))
+        s_equal = self.decode(self.sv.Struct(self.ns.S), json.dumps({'f': 'F'}))
+        s_unequal = self.decode(self.sv.Struct(self.ns.S), json.dumps({'f': 'G'}))
+        self.assertEqual(s, s_equal)
+        self.assertEqual(hash(s), hash(s_equal))
+        self.assertNotEqual(s, s_unequal)
+        self.assertNotEqual(hash(s), hash(s_unequal))
+
+    def test_struct_equality_with_different_types(self):
+        """Structs of different types that do not have an inheritance relationship are not considered
+        equal to each other."""
+        s = self.decode(self.sv.Struct(self.ns.S), json.dumps({'f': 'F'}))
+        t = self.decode(self.sv.Struct(self.ns.T), json.dumps({'f': 'F'}))
+        self.assertNotEqual(s, t)
+        # They still hash to the same value, since they have the same fields, but this is
+        # fine since we don't expect to use a large number of structs as dict keys.
+        self.assertEqual(hash(s), hash(t))
+
+        # S_extend and S_extend2 are indirectly related because they both extend S, but they do not
+        # have a direct line of inheritance to each other.
+        s_extend = self.decode(self.sv.Struct(self.ns.SExtend), json.dumps({'f': 'F', 'g': 'G'}))
+        s_extend2 = self.decode(self.sv.Struct(self.ns.SExtend2), json.dumps({'f': 'F', 'g': 'G'}))
+        self.assertNotEqual(s_extend, s_extend2)
+        # They still hash to the same value, since they have the same fields, but this is
+        # fine since we don't expect to use a large number of structs as dict keys.
+        self.assertEqual(hash(s_extend), hash(s_extend2))
+
+    def test_extended_struct_equality(self):
```

Comment:
I'm not sure if we should treat a subtype and its super type with the same value in their common fields to be equal, after all a subtype has extra fields that distinguish itself from its super type.